pilot/pkg/model/endpointshards.go

```diff
@@ -94,15 +94,26 @@ func (es *EndpointShards) Keys() []ShardKey {
 
 // CopyEndpoints takes a snapshot of all endpoints. As input, it takes a map of port name to number, to allow it to group
 // the results by service port number. This is a bit weird, but lets us efficiently construct the format the caller needs.
-func (es *EndpointShards) CopyEndpoints(portMap map[string]int) map[int][]*IstioEndpoint {
+func (es *EndpointShards) CopyEndpoints(portMap map[string]int, ports sets.Set[int]) map[int][]*IstioEndpoint {
```

Comment:
can we initialize the ports sets from portMap in this function

Reply:
yeah that makes sense. I was thinking it was faster to do it this way, but it shouldn't be